python/meta.py

```diff
@@ -0,0 +1,151 @@
+import mgp
+from mage.meta.parameters import Parameter
+from collections import defaultdict
+from typing import Dict, Tuple, Union, Iterator
+
+
+NodeKeyType = Tuple[str, ...]
+RelationshipKeyType = Tuple[NodeKeyType, str, NodeKeyType]
+
+
+class Counter:
+    def __init__(self, initial_value: int = 0):
+        self.total_count = initial_value
+        self.count_by_property_name = defaultdict(int)
+
+    def increment(self) -> None:
+        self.total_count += 1
+
+    def increment_property(self, property_name: str) -> None:
+        self.count_by_property_name[property_name] += 1
+
+    def to_dict(self, include_properties):
+        return (
+            {
+                Parameter.COUNT.value: self.total_count,
+                Parameter.PROPERTIES_COUNT.value: self.count_by_property_name,
+            }
+            if include_properties
+            else {Parameter.COUNT.value: self.total_count}
+        )
+
+
+@mgp.read_proc
+def schema(
+    context: mgp.ProcCtx, include_properties: bool = False
+) -> mgp.Record(nodes=mgp.List[mgp.Map], relationships=mgp.List[mgp.Map]):
+    """
+    Procedure to generate the graph database schema.
+
+    Args:
+        context (mgp.ProcCtx): Reference to the context execution.
+        include_properties (bool): If set to True, the graph schema will include properties count information.
+
+    Returns:
+        mgp.Record containing a mgp.List of mgp.Map objects representing nodes in the graph schema and a mgp.List of mgp.Map objects representing relationships.
+
+    Example:
+        Get graph schema without properties count:
+            `CALL meta.schema() YIELD nodes, relationships RETURN nodes, relationships;`
+        Get graph schema with properties count:
+            `CALL meta.schema(true) YIELD nodes, relationships RETURN nodes, relationships;`
+    """
+
+    if len(context.graph.vertices) == 0:
+        raise Exception(
+            "Can't generate a graph schema since there is no data in the database."
+        )
```

Comment:
This might be a problem if the graph is huge. Because, what will happen here, is that the iterator needs to go through each node, and that means making calls to the Memgraph database each time to get a vertex, creating one as Python, and destroying afterward. You can do the following here. Instead of this chunk of code, you have for loop already going over nodes. Add some kind of counter inside that loop. Or use enumerate function. If at the end "i" is 0, throw an error then. I know this is better practice, but I am here talking about optimization.

Reply:
Cool, makes sense, I will add this :)